osu.Game/Rulesets/Edit/Checks/CheckInconsistentSettings.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckInconsistentSettings : ICheck
+    {
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Settings, "Inconsistent settings", CheckScope.BeatmapSet);
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateInconsistentSetting(this, IssueType.Warning),
+            new IssueTemplateInconsistentSetting(this, IssueType.Negligible),
+        };
+
+        public IEnumerable<Issue> Run(BeatmapVerifierContext context)
+        {
+            var difficulties = context.BeatmapsetDifficulties;
+
+            if (difficulties.Count <= 1)
+                yield break;
+
+            var referenceBeatmap = context.Beatmap;
+
+            // Define fields to check
+            var fieldsToCheck = new (IssueType issueType, string fieldName, Func<IBeatmap, string> fieldSelector)[]
+            {
+                (IssueType.Warning, "Audio lead-in", b => b.AudioLeadIn.ToString(CultureInfo.InvariantCulture)),
+                (IssueType.Warning, "Countdown", b => b.Countdown.ToString()),
+                (IssueType.Warning, "Countdown offset", b => b.CountdownOffset.ToString()),
+                (IssueType.Warning, "Epilepsy warning", b => b.EpilepsyWarning.ToString()),
+                (IssueType.Warning, "Letterbox during breaks", b => b.LetterboxInBreaks.ToString()),
+                (IssueType.Warning, "Samples match playback rate", b => b.SamplesMatchPlaybackRate.ToString()),
+                (IssueType.Warning, "Widescreen support", b => b.WidescreenStoryboard.ToString()),
+                (IssueType.Negligible, "Tick Rate", b => b.Difficulty.SliderTickRate.ToString(CultureInfo.InvariantCulture)),
```

Comment:
going through `ToString()` to check equality of *anything* is criminal, I cannot simply ignore this.

please refactor such that this operates on actual pre-stringed values. will likely involve something like rearranging the loops below so that the inner loop is across difficulties, and some degree of generics usage. probably something along the lines of

```csharp
void checkIssue<T>(IssueType issueType, string fieldName, Func<IBeatmap, T> fieldSelector)
    where T : IEquatable<T>
{
    // ...
}
```

or similar

Reply:
Better but not quite. `EqualityComparer<object>.Default` is generally not advised for usage.

Please check https://github.com/ppy/osu/pull/34540/commits/c2a85032a6e06785a43f74624a384410141ed7de and whether it behaves how you'd expect it to. It passes tests so I'd like to think so but I'd rather you check it anyway.